src/domain/service/fileUploader.service.ts

```diff
@@ -124,17 +147,38 @@ export default class FileUploaderService {
     return fileData;
   }
 
+
+  /**
+   * Validate file location due to passed file type
+   *
+   * @param type - passed file type
+   * @param location - location object to check
+   */
+  private validateLocation(type: FileTypes, location: Location): void {
+    switch (type) {
+      /**
+       * Check location, if file is note attachment, noteId is required
+       */
+      case FileTypes.NoteAttachment:
+        if (isEmpty((location as NoteAttachmentFileLocation).noteId)) {
+          throw new DomainError('Invalid location for passed file type');
+        }
+    };
```

Comment:
I'd suggest to invert validation logic: 

If all requirements met, just "return". Otherwise throw.

Right now it won't throw in case of wrong type and location

Reply:
done